common/metadata-helper.cpp

```diff
@@ -0,0 +1,324 @@
+#include "metadata-helper.h"
+
+#ifdef WIN32
+
+#include <Windows.h>
+#include <memory>
+#include <vector>
+#include <algorithm>
+#include <iterator>
+
+#include <regex>
+#include <iostream>
+#include <sstream>
+#include <functional>
+#include <map>
+#include <memory>
+
+#include <librealsense2/rs.hpp>
+
+#define MAX_KEY_LENGTH 255
+#define MAX_VALUE_NAME 16383
+
+#endif
+
+namespace rs2
+{
+#ifdef WIN32
+    struct device_id
+    {
+        std::string pid, mi, guid, sn;
+    };
+
+    inline std::string to_lower(std::string x)
+    {
+        transform(x.begin(), x.end(), x.begin(), tolower);
+        return x;
+    }
+
+    bool operator==(const device_id& a, const device_id& b)
+    {
+        return (to_lower(a.pid) == to_lower(b.pid) &&
```

Comment:
strcasecmp() would compare just as well without all the extra allocations/transformations